kpr.cpp

```
@@ -0,0 +1,32 @@
+
+    char buf[256] = { '\\0' };
+    char key[32] = { '\\0' };
+    int index = 0;
+    for (int i = 0; i < nInnerCount; ++i) {
+    (void)snprintf(key, sizeof(key), CONFIG_ITEM_DNSCHK, i);
+    if (0 == iniInner.GetLong(section, key, 1)) {
+        continue;
+    }
+
+    memset(buf, 0, sizeof(buf));
+    (void)snprintf(key, sizeof(buf), CONFIG_ITEM_DNS, i);
```

cppcheck (2.10-2)

[error] Buffer is accessed out of bounds

(bufferAccessOutOfBounds)